requirements-data_and_models.txt

```diff
@@ -24,7 +24,7 @@ https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.4.0/en_ner_craft_
 https://s3-us-west-2.amazonaws.com/ai2-s2-scispacy/releases/v0.4.0/en_ner_jnlpba_md-0.4.0.tar.gz
 matplotlib==3.4.1
 scipy==1.6.2
-transformers==4.5.1
+transformers==3.4.0
```

Comment:
Unfortunately, it is raising an issue, see below:
```
06/29/2021 13:37:16 - WARNING - __main__ -   Process rank: -1, device: cuda:0, n_gpu: 6, distributed training: False, 16-bits training: False
06/29/2021 13:37:16 - INFO - __main__ -   Training/evaluation parameters TrainingArguments(output_dir=biobert_cord19_v1, overwrite_output_dir=False, do_train=True, do_eval=True, do_predict=False, evaluation_strategy=IntervalStrategy.STEPS, prediction_loss_only=False, per_device_train_batch_size=16, per_device_eval_batch_size=16, gradient_accumulation_steps=1, eval_accumulation_steps=None, learning_rate=5e-05, weight_decay=0.0, adam_beta1=0.9, adam_beta2=0.999, adam_epsilon=1e-08, max_grad_norm=1.0, num_train_epochs=3.0, max_steps=-1, lr_scheduler_type=SchedulerType.LINEAR, warmup_ratio=0.0, warmup_steps=0, logging_dir=biobert_cord19_v1__logs, logging_strategy=IntervalStrategy.STEPS, logging_first_step=False, logging_steps=5000, save_strategy=IntervalStrategy.STEPS, save_steps=5000, save_total_limit=20, no_cuda=False, seed=42, fp16=False, fp16_opt_level=O1, fp16_backend=auto, fp16_full_eval=False, local_rank=-1, tpu_num_cores=None, tpu_metrics_debug=False, debug=False, dataloader_drop_last=False, eval_steps=5000, dataloader_num_workers=16, past_index=-1, run_name=biobert_cord19_v1, disable_tqdm=False, remove_unused_columns=True, label_names=None, load_best_model_at_end=False, metric_for_best_model=None, greater_is_better=None, ignore_data_skip=False, sharded_ddp=[], deepspeed=None, label_smoothing_factor=0.0, adafactor=False, group_by_length=False, length_column_name=length, report_to=[], ddp_find_unused_parameters=None, dataloader_pin_memory=True, skip_memory_metrics=False, _n_gpu=6, mp_parameters=)
/home/edelattr/.local/lib/python3.7/site-packages/transformers/models/auto/modeling_auto.py:762: FutureWarning: The class `AutoModelWithLMHead` is deprecated and will be removed in a future version. Please use `AutoModelForCausalLM` for causal language models, `AutoModelForMaskedLM` for masked language models and `AutoModelForSeq2SeqLM` for encoder-decoder models.
  FutureWarning,
Some weights of BertForMaskedLM were not initialized from the model checkpoint at clagator/biobert_v1.1_pubmed_nli_sts and are newly initialized: ['cls.predictions.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.decoder.weight', 'cls.predictions.transform.LayerNorm.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
Traceback (most recent call last):
  File "train.py", line 334, in <module>
    main()
  File "train.py", line 255, in main
    data_args.block_size = tokenizer.max_len
AttributeError: 'BertTokenizerFast' object has no attribute 'max_len'
```

Reply:
> Unfortunately, it is raising an issue

Is everything working when changing `max_len` to `model_max_length` in `train.py`?

It would be the best to do at the moment, IMO. Indeed, we don't have currently a way to have dependency versions specific to only one pipeline.

Usually, we can change the dependency versions without caring. It is because we can checkout a previous commit and this commit has the dependency versions we used for a given pipeline.

🚫 Here we are in other case: decreasing dependency versions. That's a issue.

> Which version did you use in the past ?

v3.4.0, said [here](https://github.com/BlueBrain/Search/blob/12988ef564dd4e6373a7455f5ee30c0608e2e972/data_and_models/pipelines/sentence_embedding/scripts/build.sh#L22).

Reply:
> Indeed, we don't have currently a way to have dependency versions specific to only one pipeline.

I see. Could we maybe force this version of transformers in the Dockerfile specific for `sentence_embedding` pipeline then ? Indeed, I would be in favor of keeping the good version of the script and keep it as-is. 

Another solution could also be to upgrade the script itself, indeed I saw that `transformers` uploaded a new `language_modeling` script compatible with versions >= 4.0.0.

What do you think ?

Reply:
Have you tired the following?

> Is everything working when changing `max_len` to `model_max_length` in `train.py`?

Reply:
It is raising other issues:

```Traceback (most recent call last):
  File "train.py", line 334, in <module>
    main()
  File "train.py", line 263, in main
    get_dataset(data_args, tokenizer=tokenizer, cache_dir=model_args.cache_dir) if training_args.do_train else None
  File "train.py", line 165, in get_dataset
    return _dataset(args.train_data_file)
  File "train.py", line 150, in _dataset
    return LineByLineTextDataset(tokenizer=tokenizer, file_path=file_path, block_size=args.block_size)
  File "/home/edelattr/.local/lib/python3.7/site-packages/transformers/data/datasets/language_modeling.py", line 135, in __init__
    batch_encoding = tokenizer(lines, add_special_tokens=True, truncation=True, max_length=block_size)
  File "/home/edelattr/.local/lib/python3.7/site-packages/transformers/tokenization_utils_base.py", line 2271, in __call__
    **kwargs,
  File "/home/edelattr/.local/lib/python3.7/site-packages/transformers/tokenization_utils_base.py", line 2456, in batch_encode_plus
    **kwargs,
  File "/home/edelattr/.local/lib/python3.7/site-packages/transformers/tokenization_utils_fast.py", line 382, in _batch_encode_plus
    pad_to_multiple_of=pad_to_multiple_of,
  File "/home/edelattr/.local/lib/python3.7/site-packages/transformers/tokenization_utils_fast.py", line 335, in set_truncation_and_padding
    self._tokenizer.enable_truncation(max_length, stride=stride, strategy=truncation_strategy.value)
OverflowError: int too big to convert
```

If I change tokenizer.model_max_length to 512,

```
Traceback (most recent call last):
  File "train.py", line 334, in <module>
    main()
  File "train.py", line 288, in main
    prediction_loss_only=True,
TypeError: __init__() got an unexpected keyword argument 'prediction_loss_only'
```

Moreover, launching the script is also creating this warning:
```
FutureWarning: This dataset will be removed from the library soon, preprocessing should be handled with the 🤗 Datasets library. You can have a look at this example script for pointers: https://github.com/huggingface/transformers/blob/master/examples/language-modeling/run_mlm.py
```

Reply:
> > Indeed, we don't have currently a way to have dependency versions specific to only one pipeline.
> 
> I see. Could we maybe force this version of transformers in the Dockerfile specific for `sentence_embedding` pipeline then ? Indeed, I would be in favor of keeping the good version of the script and keep it as-is.

I would be in favour of the above and creating an issue for doing the following:

> Another solution could also be to upgrade the script itself, indeed I saw that `transformers` uploaded a new `language_modeling` script compatible with versions >= 4.0.0.

Indeed, migrating from transformers v3 to v4 is outside the scope of the ticket. That would add a lot of work too.

Reply:
Done in d13e57b.